source/docs/network_filter_fuzzing.md

```diff
@@ -0,0 +1,116 @@
+        "//test/config:utility_lib",
+    ],
+)
+```
+## Step2. Add the filter name into supported_filter_names
+In [uber_per_writefilter.cc](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/uber_per_writefilter.cc), add the filter name into the vector `supported_filter_names` in method `UberWriteFilterFuzzer::filterNames()`.
+```
+const std::vector<absl::string_view> supported_filter_names = {
+      ...
+      NetworkFilterNames::get().ExtAuthorization, NetworkFilterNames::get().TheNewFilterCreatedByYou,
+      ...
+    };
+```
+
+# Add test cases into corpus
+Good test cases can provide good examples for fuzzers to find more paths in the code, increase the coverage and help find bugs more efficiently.
+Each test case is a file under the folder [network_readfilter_corpus](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_readfilter_corpus) or [network_writefilter_corpus](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_writefilter_corpus). It consists of two parts: `config` and `actions`. 
+`config` is the protobuf to instantiate a filter, and `actions` are sequences of actions to take in order to test the filter. 
+An example for testing MongoProxy filter:
+```
+config {
+  name: "envoy.filters.network.mongo_proxy"
+  typed_config {
+    type_url: "type.googleapis.com/envoy.extensions.filters.network.mongo_proxy.v3.MongoProxy"
+    value: "\n\001\\\032\t\032\002\020\010\"\003\010\200t \001"
+  }
+}
+actions {
+  on_new_connection {
+  }
+}
+actions {
+  on_data {
+    data: "\120\0\0\0\1\0\0\0\1\0\0\0\324\7\0\0\4\0\0\0\164\145\163\164\56\164\145\163\164\0\24\0\0\0\377\377\377\377\52\0\0\0\2\163\164\162\151\156\147\137\156\145\145\144\137\145\163\143\0\20\0\0\0\173\42\146\157\157\42\72\40\42\142\141\162\12\42\175\0\0"
+  }
+}
+actions {
+  advance_time {
+    milliseconds: 10000
+  }
+}
+```
+`config.name` is the name of the filter. 
+`config.typed_config.type_url` is the type url of the filter config API. 
+`config.typed_config.value` is the serialized string of the config protobuf, and in C++ we can call`config.SerializeAsString()` to obtain this. This string may contain special characters. Recommend using octal or hexadecimal sequence for the string.
+`actions.on_data.data` is the buffer parameter `data`(in string format) for testing ReadFilter's method onData(). This string may contain special characters. Recommend using octal or hexadecimal sequence for the string.
+`actions.on_data.end_stream` is the bool parameter `end_stream` for testing ReadFilter's method onData().
+`actions.on_write.data` is the buffer parameter `data`(in string format) for testing WriteFilter's method onWrite(). This string may contain special characters. Recommend using octal or hexadecimal sequence for the string.
+`actions.on_write.end_stream` is the bool parameter `end_stream` for testing WriteFilter's method onWrite().
```

Comment:
maybe make it clear that this only belongs to writefilter fuzz testcase? (Or combine the `data` and `end_stream` for both filters and specify it like:
* `actions.on_data.data.` or `actions.on_write.data` is the buffer parameter `data` for testing ReadFilter's (or WriteFilter's`)

Reply:
> maybe make it clear that this only belongs to writefilter fuzz testcase? (Or combine the `data` and `end_stream` for both filters and specify it like:
> 
> * `actions.on_data.data.` or `actions.on_write.data` is the buffer parameter `data` for testing ReadFilter's (or WriteFilter's`)

Combined. Does it look better?